controlplane/kubeadm/internal/controllers/upgrade.go

```diff
@@ -74,6 +74,16 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 		return ctrl.Result{}, errors.Wrap(err, "failed to update the kubernetes version in the kubeadm config map")
 	}
 
+	// FIXME(chrischdi): use a better semver comparison
+	// FIXME(chrischdi): think about having a better if structure 😅
```

Comment:
wondering if we can combine above change with the next one, so we update the config map only once 🤔